system/lib/wasmfs/backends/node_backend.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <memory>
+
+#include "backend.h"
+#include "file.h"
+#include "wasmfs.h"
+
+namespace wasmfs {
+
+class NodeBackend;
+
+extern "C" {
+
+// Fill `entries` and return 0 or an error code.
+int _wasmfs_node_readdir(const char* path,
+                         std::vector<Directory::Entry>* entries);
+// Write `mode` and return 0 or an error code.
+int _wasmfs_node_get_mode(const char* path, mode_t* mode);
+
+// Create a new file system entry and return 0 or an error code.
+int _wasmfs_node_insert_file(const char* path,
+                             mode_t mode); // fs.openSync(name, flags, mode)
+int _wasmfs_node_insert_directory(
+  const char* path, mode_t mode); // fs.mkdirSync(name, { mode: ... })
+
+} // extern "C"
+
+class NodeFile : public DataFile {
+public:
+  std::string path;
```

Comment:
Which path is this?

A comment on this class would be good to explain the design.

Reply:
This is the path on the underlying Node file system. I'll add comments here and in `NodeDirectory`.